src/v/cluster/tm_stm.cc

```diff
@@ -280,15 +284,15 @@ ss::future<tm_stm::op_status> tm_stm::re_register_producer(
     tx.status = tm_transaction::tx_status::ready;
     tx.pid = pid;
     tx.tx_seq += 1;
-    tx.etag = _insync_term;
+    tx.etag = expected_term;
```

Comment:
above in tx_reset_ready the tag is left unchanged. is this not the case here because it is an initialization case?

Reply:
`reset_tx_ready` also sets `etag` to term which in one branch is `expected_term`